### DISCUSSION TITLE
Composer package for the binary

### DISCUSSION THREAD

**Discussion by binaryfire** [php/frankenphp#502] [Ideas]
Hi @dunglas 

I need a way of locking the FrankenPHP binary version with each commit so I can have reproducible CI pipeline builds. Without that, if I roll back to a commit from a week ago and the binary has been updated since then, a newer binary will be deployed with the old commit (I'm using Octane). Which isn't what you want in production environments.

What do you think about releasing the binary as a composer package? That would make it very easy to version control per-project.

The only alternative I can think of is committing the binary in every project but that'd greatly increase the size of all my repos.

**Comment by jwhulette** [comment]
We use docker builds and this is what I currently use,
```
ENV FRANKENPHP_VERSION="v1.2.2"

RUN if [ "$APP_ENV" = "prod" ] ; \
    then curl -L https://github.com/dunglas/frankenphp/releases/download/$FRANKENPHP_VERSION/frankenphp-linux-aarch64 --output frankenphp && \
             chmod 755 frankenphp && \
             chmod u+x frankenphp; \
fi
```
Not the most elegant but it works
